vmm/src/memory_manager.rs

```diff
@@ -103,6 +110,34 @@ impl MemoryZone {
     pub fn virtio_mem_zone(&self) -> &Option<VirtioMemZone> {
         &self.virtio_mem_zone
     }
+    pub fn virtio_mem_zone_mut(&mut self) -> &mut Option<VirtioMemZone> {
+        &mut self.virtio_mem_zone
+    }
+    pub fn active_ranges(&self) -> Vec<(u64, u64, u64)> {
+        let mut ranges = Vec::new();
```

Comment:
You define a new Range type and then don't use it here...is that because it's not right. I would expect we could return a useful vector of structs here.

Reply:
Oh good catch, I forgot to take advantage of it here :)